player-service/src/main/java/org/libertybikes/player/service/RankingService.java

```diff
@@ -69,7 +92,31 @@ public long getRank(@PathParam("playerId") String id) {
 
     @POST
     @Path("/{playerId}/recordGame")
-    public void recordGame(@PathParam("playerId") String id, @QueryParam("place") int place) {
+    public void recordGame(@PathParam("playerId") String id, @QueryParam("place") int place, @QueryParam("jwt") String jwt) {
+
+        if (signingKey == null) {
+            try {
+                getKeyStoreInfo();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        try {
+
+            Jwts.parser().setSigningKey(signingKey).parseClaimsJws(jwt);
```

Comment:
This is using JWT, but it's not using MP Jwt.  With MP Jwt the only thing player service should have to do is add 1 annotation for roles allowed like this:
```java
@POST
@Path("/{playerId}/recordGame")
@RolesAllowed({ "system" })
public void recordGame(@PathParam("playerId") String id, @QueryParam("place") int place) {	
 // ...
```
as well as a single server.xml entry like this:
```xml
<mpJwt id="jwtUserConsumer" keyName="default" audiences="simpleapp" issuer="${jwt.issuer}"/>
```

The player service should not have to directly interact with the io.jsonwebtoken library or load/read the keystore.  Perhaps have a look at the MP Jwt guide for example: https://openliberty.io/guides/microprofile-jwt.html